cpp/src/arrow/parquet/writer.cc

```diff
@@ -143,6 +153,82 @@ Status FileWriter::Close() {
 
 FileWriter::~FileWriter() {}
 
+// Create a slice of a PrimitiveArray.
+//
+// This method is specially crafted for WriteFlatTable and assumes the following:
+//  * chunk_size is a multiple of 512
```

Comment:
`DCHECK` sadly does not work currently in the `arrow::parquet` namespace, thus I haven't added one.

For the chunk_size: This assumption is mainly so we have 64byte alignment in all sliced arrays, even the null bitmap. @emkornfield do you think we should respect that here also or just limit the chunk size so that we can easily slice the null-bitmap?